### DISCUSSION TITLE
MacOS won't allow Desktop Remote when not actively logged in

### DISCUSSION THREAD

**Discussion by rrmoore64** [Ylianst/MeshCentral#6207] [Issues]
I have been using Mesh Central for almost 2 years now and have followed the thread about the newer MacOS issues closely.  I have a pretty good process for getting it working consistently on Sonoma and Ventura on both Intel and Apple Silicon architecture.  One issue I cannot seem to figure out - and that I have not seen discussed;
If the Mac running the meshagent is NOT logged in - either sitting at the login screen after a logout or after a simple screen lock - Mesh will connect to the desktop; but will not show the screen nor allow any interaction if I type at all.  I can't find a way to trigger it to log in as a user nor any way to get it to show the screen and allow me to log in.
Thoughts from anyone?  I know the product is very sparsely supported for the past many months - but thought maybe someone has run into this.
Thanks, 
Randy

**Comment by si458** [comment]
Sadly it's not possible!
We need a developer who can figure out how to do this and implement it.

I figured out how to wake up the screen to the login window when it's locked, but this only works IF u log in/autologon to begin with.

So when your mac first starts up, the login screen is behind lock and key, and we can't get access to the display.

So sadly the only way around this at the moment is to autologin as a user when the mac restarts

**Comment by rrmoore64** [comment]
Well, at least it is not just me.  Thanks!

When will Apple every stop fighting us?!

**Reply by si458** [reply]
Couldn't agree more!

**Reply by rrmoore64** [reply]
Although, there are a few remote products that DO work.  Teamviewer, Jump Desktop, and Apple's Remote Desktop all seem to allow login to the GUI - so it must be "possible"

**Comment by silversword411** [comment]
SC connects outside of a logged in user. There's 2 .plist files under /Library/LaunchAgents

Hopefully this helps someone

<img width="925" alt="2024-07-22_224851 - Code_●_name-sid-prelogin plist_↔_name-sid-onlogin plist" src="https://github.com/user-attachments/assets/e57c90d6-69f8-4b3d-952d-caae8c762111">
